### PR TITLE
People icon in an Adaptive Card

### PR DESCRIPTION
**Pull request by Harikrishnan-MSFT**

### CODE REVIEW

**Comment by Harikrishnan-MSFT** [comment]
**Build Pipeline - **

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs [hunk 1]
```diff
@@ -151,5 +151,43 @@ public static Attachment sendCardWithEmoji()
             return emojiAdaptiveCardAttachment;
         }
 
+        /// <summary>
+        /// Persona card Icon in an Adaptive Card
+        /// If you want to show a single user in an Adaptive Card, the Adaptive Card displays the people icon and the name of the user.
+        /// </summary>
+        /// <returns>Return Microsoft.Bot.Schema.Attachment results.</returns>
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:158]: Add proper spacing.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs [hunk 2]
```diff
@@ -151,5 +151,43 @@ public static Attachment sendCardWithEmoji()
...
+        public static Attachment sendPersonaCardIcon()
+        {
+            var paths = new[] { ".", "Resources", "adaptivePeoplePersonaCardIcon.json" };
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:161]: Add proper line spacing.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs [hunk 3]
```diff
@@ -151,5 +151,43 @@ public static Attachment sendCardWithEmoji()
...
+            };
+
+          return PersonaCardIconAdaptiveCardAttachment;
+        }
+
+        /// <summary>
+        /// Persona Card Set Icon in an Adaptive Card
+        /// If you want to show multiple users in an Adaptive Card, the Adaptive Card displays only the people icon of the users.
+        /// </summary>
+        /// <returns>Return Microsoft.Bot.Schema.Attachment results.</returns>
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:177]: Add proper spacing.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs [hunk 4]
```diff
@@ -151,5 +151,43 @@ public static Attachment sendCardWithEmoji()
...
+        public static Attachment sendPersonaCardSetIcon()
+        {
+            var paths = new[] { ".", "Resources", "adaptivePeoplePersonaCardSetIcon.json" };
+            var adaptiveCardPersonaCardSetIconJson = File.ReadAllText(Path.Combine(paths));
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:181]: Add proper line spacing.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs [hunk 5]
```diff
@@ -151,5 +151,43 @@ public static Attachment sendCardWithEmoji()
...
+        /// </summary>
+        /// <returns>Return Microsoft.Bot.Schema.Attachment results.</returns>
+        public static Attachment sendPersonaCardIcon()
+        {
+            var paths = new[] { ".", "Resources", "adaptivePeoplePersonaCardIcon.json" };
+            var adaptiveCardPersonaCardIconJson = File.ReadAllText(Path.Combine(paths));
+
+            var PersonaCardIconAdaptiveCardAttachment = new Attachment()
+            {
+                ContentType = "application/vnd.microsoft.card.adaptive",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:166]: If we are using it at multiple places, please create a constant in same file.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Dialogs/MainDialog.cs [hunk 6]
```diff
@@ -105,13 +105,23 @@ private async Task<DialogTurnResult> SelectCardAsync(WaterfallStepContext stepCo
                     // Sends Card With Emoji
                     reply.Attachments.Add(AllCards.sendCardWithEmoji());
                     break;
+                case "Persona":
+                    // Sends Persona Card Icon
+                    reply.Attachments.Add(AllCards.sendPersonaCardIcon());
+                    break;
+                case "PersonaSet":
+                    // Sends Persona Card Set Icon
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Dialogs/MainDialog.cs:113]: // Sends Persona Card Set Icons

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Dialogs/MainDialog.cs [hunk 7]
```diff
@@ -105,13 +105,23 @@ private async Task<DialogTurnResult> SelectCardAsync(WaterfallStepContext stepCo
...
+                    break;
                 default:
                     reply.Attachments.Add(AllCards.sendInfoMasking());
                     reply.Attachments.Add(AllCards.sendFullWidthCardAdaptiveCard());
                     reply.Attachments.Add(AllCards.sendStageViewImagesCard());
                     reply.Attachments.Add(AllCards.sendOverFlowMenuCard());
                     reply.Attachments.Add(AllCards.sendHTMLConnectorCard());
                     reply.Attachments.Add(AllCards.sendCardWithEmoji());
+                    reply.Attachments.Add(AllCards.sendPersonaCardIcon());
+                    reply.Attachments.Add(AllCards.sendPersonaCardSetIcon());
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Dialogs/MainDialog.cs:124]: Icons

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
+﻿{
+  "$schema": "https://adaptivecards.io/schemas/adaptive-card.json",
+  "type": "AdaptiveCard",
+  "version": "1.0.0",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json:4]: Please confirm with engineering team whether we need to use version 1.0.0 or the latest one.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+      "weight": "bolder"
+    },
+    {
+      "type": "Component",
+      "name": "graph.microsoft.com/user",
+      "view": "compact",
+      "properties": {
+        "id": "{{Object-ID}}",
+        "displayName": "{{User-Display-Name}}",
+        "userPrincipalName": "{{User-Principal-Name}}"
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json:18]: Make sure to add the details in Readme for updating these data.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+      "type": "TextBlock",
+      "text": "Persona",
+      "weight": "bolder"
+    },
+    {
+      "type": "Component",
+      "name": "graph.microsoft.com/user",
+      "view": "compact",
+      "properties": {
+        "id": "{{Object-ID}}",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json:16]: What is object id?

Add proper

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json (hunk 9 above)
Comment (ChetanSharma-msft) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json:18]: Same comments applicable to other Json card as well.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardSetIcon.json [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+            "id": "{{Object-ID}}",
+            "displayName": "{{User-Display-Name}}",
+            "userPrincipalName": "{{User-Principal-Name}}"
+          },
+          {
+            "id": "{{Object-ID}}",
+            "displayName": "{{User-Display-Name}}",
+            "userPrincipalName": "{{User-Principal-Name}}"
+          }
+        ]
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardSetIcon.json:27]: Check above card comments.

File: samples/bot-formatting-cards/csharp/README.md [hunk 12]
```diff
@@ -125,6 +125,46 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
             }
             ]
         ```
+   
+   **Note: The users are in the card component have to be on the same tenant.**
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/README.md:129]: Mention properly that, In Adaptive Card what we are defining (User details) should be exist in the same tenant where you are testing the app (teams login) etc...

File: samples/bot-formatting-cards/csharp/README.md [hunk 13]
```diff
@@ -125,6 +125,46 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
...
+    - Navigate to samples\bot-formatting-cards\csharp\BotFormattingCards\Resources\adaptivePeoplePersonaCardIcon.json
+      1) On line 16, replace {{Object-ID}}  
+      2) On line 17, replace {{User-Display-Name}}
+      3) On line 18, replace {{User-Principal-Name}}
+
+        - E.g. 
+        ```
+        "properties": {
+        "id": "65f50003-e15d-434a-9e14-0fcfeb3d7817",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/README.md:138]: Please don't put exact values.

File: samples/bot-formatting-cards/csharp/README.md [hunk 14]
```diff
@@ -125,6 +125,46 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
...
+        "displayName": "Daniela Mandera",
+        "userPrincipalName": "damandera@microsoft.com"
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/README.md:140]: Dummy data please

File: samples/bot-formatting-cards/csharp/README.md [hunk 15]
```diff
@@ -125,6 +125,46 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
...
+        "users": [
+          {
+            "id": "65f50003-e15d-434a-9e14-0fcfeb3d7817",
+            "displayName": "Daniela Mandera",
+            "userPrincipalName": "damandera@microsoft.com"
+          },
+          {
+            "id": "65f50003-e15d-434a-9e14-0fcfeb3d7817",
+            "displayName": "Daniela Mandera",
+            "userPrincipalName": "damandera@microsoft.com"
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/README.md:163]: Please remove exact values, don't use @microsoft.com

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 16]
```diff
@@ -113,6 +114,46 @@ Run ngrok - point to port 3978
...
+      2) On line 17, replace {{User-Display-Name}}
+      3) On line 18, replace {{User-Principal-Name}}
+
+        - E.g. 
+        ```
+        "properties": {
+        "id": "65f50003-e15d-434a-9e14-0fcfeb3d7817",
+        "displayName": "Daniela Mandera",
+        "userPrincipalName": "damandera@microsoft.com"
+         }
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:129]: Please remove these valuies or use some dummy data. refer other comments.

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 17]
```diff
@@ -113,6 +114,46 @@ Run ngrok - point to port 3978
...
+        ```
+  - Navigate to samples\bot-formatting-cards\nodejs\resources\adaptivePeoplePersonaCardSetIcon.json
+      1) On line 18, replace {{Object-ID}}
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:132]: What is object id?

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 18]
```diff
@@ -113,6 +114,46 @@ Run ngrok - point to port 3978
...
+        "users": [
+          {
+            "id": "65f50003-e15d-434a-9e14-0fcfeb3d7817",
+            "displayName": "Daniela Mandera",
+            "userPrincipalName": "damandera@microsoft.com"
+          },
+          {
+            "id": "65f50003-e15d-434a-9e14-0fcfeb3d7817",
+            "displayName": "Daniela Mandera",
+            "userPrincipalName": "damandera@microsoft.com"
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:151]: Please remove these.

File: samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardIcon.json [hunk 19]
```diff
@@ -0,0 +1,22 @@
+﻿{
+  "$schema": "https://adaptivecards.io/schemas/adaptive-card.json",
+  "type": "AdaptiveCard",
+  "version": "1.0.0",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardIcon.json:4]: Check if we can use latest version?

File: samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardIcon.json [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+      "type": "TextBlock",
+      "text": "Persona",
+      "weight": "bolder"
+    },
+    {
+      "type": "Component",
+      "name": "graph.microsoft.com/user",
+      "view": "compact",
+      "properties": {
+        "id": "{{Object-ID}}",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardIcon.json:16]: object id ?

File: samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardSetIcon.json [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+      "weight": "bolder"
+    },
+    {
+      "type": "Component",
+      "name": "graph.microsoft.com/users",
+      "view": "compact",
+      "properties": {
+        "users": [
+          {
+            "id": "{{Object-ID}}",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardSetIcon.json:18]: object id?

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments on your PR.

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs (hunk 1 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:158]: FIxed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs (hunk 2 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:161]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs (hunk 3 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:177]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs (hunk 4 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:181]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs (hunk 5 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Cards/AllCards.cs:166]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Dialogs/MainDialog.cs (hunk 6 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Dialogs/MainDialog.cs:113]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Dialogs/MainDialog.cs (hunk 7 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Dialogs/MainDialog.cs:124]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json (hunk 8 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json:4]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json (hunk 9 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json:18]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json (hunk 10 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardIcon.json:16]: Fixed

File: samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardSetIcon.json (hunk 11 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/BotFormattingCards/Resources/adaptivePeoplePersonaCardSetIcon.json:27]: Fixed

File: samples/bot-formatting-cards/csharp/README.md (hunk 12 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:129]: Fixed

File: samples/bot-formatting-cards/csharp/README.md (hunk 13 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:138]: Updated dummy data.

File: samples/bot-formatting-cards/csharp/README.md (hunk 14 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:140]: Updated dummy data.

File: samples/bot-formatting-cards/csharp/README.md (hunk 15 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:163]: Updated dummy data.

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 16 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:129]: Updated dummy data.

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 17 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:132]: Fixed

File: samples/bot-formatting-cards/csharp/README.md (hunk 13 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:138]: Fixed

File: samples/bot-formatting-cards/csharp/README.md (hunk 14 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:140]: Fixed

File: samples/bot-formatting-cards/csharp/README.md (hunk 15 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:163]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 16 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:129]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 18 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:151]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardIcon.json (hunk 19 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardIcon.json:4]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardIcon.json (hunk 20 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardIcon.json:16]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardSetIcon.json (hunk 21 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/adaptivePeoplePersonaCardSetIcon.json:18]: Fixed

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

File: samples/bot-formatting-cards/csharp/README.md [hunk 22]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
...
+  - Locate the Basic info section on the user's Profile page. The Object ID that is displayed is the user's unique object ID.
+    - Navigate to samples\bot-formatting-cards\csharp\BotFormattingCards\Resources\adaptivePeoplePersonaCardIcon.json
+      1) On line 16, replace {{User-Object-ID}}  
+      2) On line 17, replace {{User-Display-Name}}
+      3) On line 18, replace {{User-Principal-Name}}
+
+        - E.g. 
+        ```
+        "properties": {
+        "id": "87d349ed-e15d-434a-9e14-87d349ed",
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:139]: Fixed

File: samples/bot-formatting-cards/csharp/README.md [hunk 23]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
...
+      4) On line 23, replace {{User-Object-ID}}  
+      5) On line 24, replace {{User-Display-Name}}
+      6) On line 25, replace {{User-Principal-Name}}
+      
+        - E.g. 
+        ```
+      "properties": {
+        "users": [
+          {
+            "id": "87d349ed-e15d-434a-9e14-87d349ed",
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:157]: Fixed

File: samples/bot-formatting-cards/csharp/README.md [hunk 24]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
...
+            "displayName": "Test",
+            "userPrincipalName": "Test@w.com"
+          },
+          {
+            "id": "87d349ed-e15d-434a-9e14-87d349ed",
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:162]: Fixed

File: samples/bot-formatting-cards/csharp/README.md [hunk 25]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
...
+            "displayName": "Test",
+            "userPrincipalName": "Test@w.com"
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:164]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 26]
```diff
@@ -113,6 +114,47 @@ Run ngrok - point to port 3978
...
+        ```
+      "properties": {
+        "users": [
+          {
+            "id": "87d349ed-e15d-434a-9e14-87d349ed",
+            "displayName": "Test",
+            "userPrincipalName": "Test@w.com"
+          },
+          {
+            "id": "87d349ed-e15d-434a-9e14-87d349ed",
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:150]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 27]
```diff
@@ -113,6 +114,47 @@ Run ngrok - point to port 3978
...
+            "displayName": "Test",
+            "userPrincipalName": "Test@w.com"
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:152]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 28]
```diff
@@ -113,6 +114,47 @@ Run ngrok - point to port 3978
...
+      3) On line 20, replace {{User-Principal-Name}}
+      4) On line 23, replace {{User-Object-ID}}  
+      5) On line 24, replace {{User-Display-Name}}
+      6) On line 25, replace {{User-Principal-Name}}
+      
+        - E.g. 
+        ```
+      "properties": {
+        "users": [
+          {
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:144]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 29]
```diff
@@ -113,6 +114,47 @@ Run ngrok - point to port 3978
...
+  - Locate the Basic info section on the user's Profile page. The Object ID that is displayed is the user's unique object ID.
+    - Navigate to samples\bot-formatting-cards\nodejs\resources\adaptivePeoplePersonaCardIcon.json
+      1) On line 16, replace {{User-Object-ID}}  
+      2) On line 17, replace {{User-Display-Name}}
+      3) On line 18, replace {{User-Principal-Name}}
+
+        - E.g. 
+        ```
+        "properties": {
+        "id": "87d349ed-e15d-434a-9e14-87d349ed",
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:127]: Fixed

File: samples/bot-formatting-cards/csharp/README.md [hunk 30]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
             }
             ]
         ```
+   
+   **Note: Mention properly that, In Adaptive Card what we are defining (User details) should be exist in the same tenant where you are testing the app (teams login) etc...**
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md]: Fixed

File: samples/bot-formatting-cards/csharp/README.md [hunk 31]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
             }
             ]
         ```
+   
+   **Note: In Adaptive Card what we are defining (User details) should be exist in the same tenant where you are testing the app (teams login) etc...**
+  - Locate the Basic info section on the user's Profile page. The Object ID that is displayed is the user's unique object ID.
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/README.md:130]: The Object ID that is displayed is the user's unique object ID.

User-Object-ID

File: samples/bot-formatting-cards/csharp/README.md [hunk 32]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
             }
             ]
         ```
+   
+   **Note: In Adaptive Card what we are defining (User details) should be exist in the same tenant where you are testing the app (teams login) etc...**
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/README.md:129]: **Note: In adaptive card, what we are defining (User details) should be exist in the same tenant where you are testing the app (teams' login) etc...**

File: samples/bot-formatting-cards/csharp/README.md [hunk 33]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
...
+      "properties": {
+        "users": [
+          {
+            "id": "87d349ed-xxxx-434a-9e14-xxxx",
+            "displayName": "Joe Smith",
+            "userPrincipalName": "Joe Smith@xxxx.com"
+          },
+          {
+            "id": "87d349ed-xxxx-434a-9e14-xxxx",
+            "displayName": "Joe Smith",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/README.md:163]: Please don't put same name and details for both the entries.

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 34]
```diff
@@ -113,6 +114,47 @@ Run ngrok - point to port 3978
             ]
         ```
 
+  **Note: In Adaptive Card what we are defining (User details) should be exist in the same tenant where you are testing the app (teams login) etc...**
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:117]: **Note: In adaptive card, what we are defining (User details) should be exist in the same tenant where you are testing the app (teams' login) etc...**

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 35]
```diff
@@ -113,6 +114,47 @@ Run ngrok - point to port 3978
...
+  - Locate the Basic info section on the user's Profile page. The Object ID that is displayed is the user's unique object ID.
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:118]: Object id or User object id?

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 36]
```diff
@@ -113,6 +114,47 @@ Run ngrok - point to port 3978
...
+      "properties": {
+        "users": [
+          {
+            "id": "87d349ed-xxxx-434a-9e14-xxxx",
+            "displayName": "Joe Smith",
+            "userPrincipalName": "Joe Smith@xxxx.com"
+          },
+          {
+            "id": "87d349ed-xxxx-434a-9e14-xxxx",
+            "displayName": "Joe Smith",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:151]: Add different values.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added few more comments.

File: samples/bot-formatting-cards/csharp/README.md (hunk 31 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:130]: Fixed

File: samples/bot-formatting-cards/csharp/README.md (hunk 32 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:129]: Fixed

File: samples/bot-formatting-cards/csharp/README.md (hunk 33 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/csharp/README.md:163]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 36 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:151]: Fixed

**Comment by Harikrishnan-MSFT** [comment]
**Build Pipeline**
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/5937987175

image

**Comment by ChetanSharma-msft** [comment]
Engineering team Approved the PR:
image

File: samples/bot-formatting-cards/csharp/README.md [hunk 37]
```diff
@@ -125,6 +125,47 @@ Please find below demo manifest which is deployed on Microsoft Azure and you can
             }
             ]
         ```
+   
+   **Note: In adaptive card, what we are defining (User details) should be exist in the same tenant where you are testing the app (teams' login) etc...**
+  - Locate the Basic info section on the user's Profile page. The Object ID that is displayed is the user's unique user object id.
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/csharp/README.md:130]: Also, update the user AAD object id from your tenant AAD users in Azure portal.

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 38]
```diff
@@ -113,6 +114,47 @@ Run ngrok - point to port 3978
             ]
         ```
 
+  **Note: In adaptive card, what we are defining (User details) should be exist in the same tenant where you are testing the app (teams' login) etc...**
+  - Locate the Basic info section on the user's Profile page. The Object ID that is displayed is the user's unique user object id.
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:118]: Update this section.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 38 above)
Comment (Harikrishnan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:118]: Fixed

**Comment by Harikrishnan-MSFT** [comment]
**Engineering team Approved the PR:**
image

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Review by Mohammed-MSFT — APPROVED** [review]
Changes Looks Fine Approving It.

**Pull request merged**